client/common/cmdline.c

```diff
@@ -1826,6 +1839,123 @@ static int parse_tls_options(rdpSettings* settings, const COMMAND_LINE_ARGUMENT_
 	    return rc;
 }
 
+static int parse_gfx_options(rdpSettings* settings, const COMMAND_LINE_ARGUMENT_A* arg)
+{
+	WINPR_ASSERT(settings);
+	WINPR_ASSERT(arg);
+
+	if (!freerdp_settings_set_bool(settings, FreeRDP_SupportGraphicsPipeline, TRUE))
+		return COMMAND_LINE_ERROR;
+
+	if (arg->Value)
+	{
+		int rc = CHANNEL_RC_OK;
+		size_t count = 0;
+		char** ptr = CommandLineParseCommaSeparatedValues(arg->Value, &count);
+		if (!ptr || (count == 0))
+			rc = COMMAND_LINE_ERROR;
+		else
+		{
+			BOOL GfxH264 = FALSE;
+			BOOL GfxAVC444 = FALSE;
+			BOOL RemoteFxCodec = FALSE;
+			BOOL GfxProgressive = FALSE;
+			BOOL codecSelected = FALSE;
+
+			for (size_t x = 0; x < count; x++)
+			{
+				const char* val = ptr[x];
+#ifdef WITH_GFX_H264
+				if (_strnicmp("AVC444", val, 7) == 0)
```

Comment:
Feel free to ignore:

Same for all other `_strnicmp` calls in this PR: Might be better to use `strlen() + 1` for the size to prevent likelihood of human error here of using the wrong size. I'm fine though with the current approach.

Reply:
this check has a bug, but another one as you imagined. `_strnicmp("AVC444", val, 6) == 0` is what is desired, as `AVC444:on` and `AVC444:off` is also valid syntax.